src/osd/PG.cc

```diff
@@ -1330,8 +1330,11 @@ bool PG::sched_scrub()
 	  << (is_active() ? ") <active>" : ") <not-active>")
 	  << (is_clean() ? " <clean>" : " <not-clean>") << dendl;
   ceph_assert(ceph_mutex_is_locked(_lock));
-  ceph_assert(!is_scrubbing());
```

Comment:
In the original bug report, do you understand how `PG_STATE_SCRUBBING` got set on the PG before we did a `sched_scrub`? The bug appears to be triggered by setting `osd_max_scrubs` greater than 1, can we try to reproduce it by doing the same in our teuthology tests?

Reply:
@ronen-fr I am able to reproduce https://tracker.ceph.com/issues/50346 by setting `osd_max_scrubs=2` for the `rados/thrash` suite. See https://pulpito.ceph.com/nojha-2021-06-23_19:31:08-rados:thrash-master-distro-basic-smithi/, jobs 6187440 and 6187460